src/main/java/com/openlattice/shuttle/dates/JavaDateTimeHelper.java

```diff
@@ -1,81 +1,123 @@
 package com.openlattice.shuttle.dates;
 
 import com.openlattice.shuttle.util.Cached;
+import com.openlattice.shuttle.util.Constants;
 import org.apache.commons.lang3.StringUtils;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
+import java.sql.Timestamp;
 import java.time.LocalDate;
 import java.time.LocalDateTime;
 import java.time.LocalTime;
 import java.time.OffsetDateTime;
 import java.time.format.DateTimeFormatter;
 import java.time.format.DateTimeParseException;
+import java.util.Optional;
 import java.util.TimeZone;
 import java.util.concurrent.ExecutionException;
 import java.util.function.BiFunction;
 
 public class JavaDateTimeHelper {
-    private static final Logger logger = LoggerFactory.getLogger(JavaDateTimeHelper.class);
+    private static final Logger logger = LoggerFactory.getLogger( JavaDateTimeHelper.class );
 
-    private final TimeZone      tz;
-    private final String[]      datePatterns;
+    private final Optional<TimeZone> tz;
+    private final String[]           datePatterns;
 
-    public JavaDateTimeHelper(TimeZone tz, String... datePatterns) {
+    public JavaDateTimeHelper( Optional<TimeZone> tz, String... datePatterns ) {
         this.tz = tz;
         this.datePatterns = datePatterns;
     }
 
-    private boolean shouldIgnoreValue(String date) {
-        return date == null || StringUtils.isBlank(date) || date.equals("NULL");
+    private boolean shouldIgnoreValue( String date ) {
+        return date == null || StringUtils.isBlank( date ) || date.equals( "NULL" );
```

Comment:
you can remove the `date == null` case -- `StringUtils.isBlank( date )` will return true if date is null

Reply:
also date being "NULL" seems like a very specific case -- it might be worth moving that into a separate filtering step for the specific integration affected

Reply:
Removed the `date == null`.  

Missing data is very common, so I don't think we should add a null-check to every integration/property.  Null in = null out.
On the other hand, some transformations replace nulls with values, so it's not general enough to ignore missing values to add this into the TransformValueMapper.

Reply:
Oh sorry I didn't mean you should skip the null check altogether (and StringUtils.isBlank will handle the case where date is actually null). I meant the case where date is the string "NULL" (the last case here) -- I'm guessing some customer gave us a csv where they had somehow transformed null values to the string "NULL" which is definitely something we should parse out for that customer, but I don't think that parsing belongs in the transform used by everybody.